storage/allocator.go

```diff
@@ -93,22 +65,16 @@ func (sl *storeList) Add(s *proto.StoreDescriptor) {
 type allocator struct {
 	sync.Mutex
```

Comment:
Not a dumb question.  I was going to try to remove it in a small follow up PR.  But I just didn't want to pollute this one any further.  I'll add a todo to consider its removal.